### DISCUSSION TITLE
`field_validator` access to other parameters

### DISCUSSION THREAD

**Discussion by hliebert** [pydantic/pydantic#7247] [Question] [answered]
I'm having trouble figuring out how to port validators that access other fields to V2. How would I do the following using field_validators?

```python
@validator("base_dates", always=True)
def set_base_dates(cls, v, values):
        """Initialize base dates."""
        # some validation for v here
        # ...
        return get_base_dates(values["components"]) if v is None else v
```

**Comment by harunyasar** [comment, marked as answer]
```python
@field_validator("field_name")
def validate_field(cls, input_value, values):
    ...
```

`input_value` is the value of the field that you validate, `values` is the other fields. You can reach the field values through `values.data`.

**Reply by anish-katukam** [reply]
Is this somewhere in the API reference that I missed?
